src/main/java/org/opentripplanner/middleware/models/TripMonitorNotification.java

```diff
@@ -110,4 +111,46 @@ public static TripMonitorNotification createInitialReminderNotification(
             )
         );
     }
+
+    /**
+     * Creates a notification that the leg has changed mode.
+     */
+    public static TripMonitorNotification createModeChangeNotification(Leg expectedLeg, Leg changedLeg, Locale locale) {
+        return new TripMonitorNotification(
+            NotificationType.MODE_CHANGE,
+            String.format(
+                Message.MODE_CHANGE_NOTIFICATION.get(locale),
+                expectedLeg.mode,
+                changedLeg.mode,
+                expectedLeg.from.name
+            )
+        );
+    }
+
+    /**
+     * Creates a notification that the origin or destination has changed.
+     */
+    public static TripMonitorNotification createStopChangeNotification(
```

Comment:
Could you clarify what you had in mind for these notification types? I thought we would need a notification that the person traveling has left their origin or arrived at the destination.

Reply:
Refactored and addressed in: https://github.com/ibi-group/otp-middleware/pull/275/commits/91d66190cb611283993b8e745148641efd695c5e